src/main/kotlin/pt/up/fe/ni/website/backend/model/dto/Dto.kt

```diff
@@ -0,0 +1,77 @@
+package pt.up.fe.ni.website.backend.model.dto
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.databind.ObjectMapper
+import javax.persistence.Entity
+import javax.validation.ConstraintViolationException
+import javax.validation.Validation
+import kotlin.reflect.KClass
+import kotlin.reflect.full.hasAnnotation
+import kotlin.reflect.jvm.jvmErasure
+
+open class Dto<T : Any> {
+
+    @JsonIgnore
+    private val entityClass: KClass<T>
+
+    constructor() : this(null)
+
+    constructor(conversionClass: KClass<T>?) {
+        this.entityClass = DtoReflectionUtils.getTypeConversionClassWithCache(this::class, conversionClass)
+    }
```

Comment:
```suggestion
    constructor(conversionClass: KClass<T>? = null) {
        this.entityClass = DtoReflectionUtils.getTypeConversionClassWithCache(this::class, conversionClass)
    }
```
Am I wrong to assume this is the same? Seems simpler to me

Reply:
At runtime, it is the same, but for IDEs it's not. The current implementation shows two possible constructors, one with no arguments and one with a class for an argument. The suggested implementation only shows one option. Since the purpose of the class argument was a bit confusing already, I chose the two constructors option, but do let me know if you want me to change it.

Reply:
Got it, it's ok like this